src/components/inputs/localized-text-input/localized-text-input.spec.js

```diff
@@ -1,267 +1,263 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import PropTypes from 'prop-types';
+import { render, fireEvent } from '../../../test-utils';
 import LocalizedTextInput from './localized-text-input';
+import { getName } from '../../../utils/localized';
 
-const createTestProps = custom => ({
-  value: { en: 'Horse', de: 'Pferd' },
-  onChange: jest.fn(),
-  selectedLanguage: 'en',
-  ...custom,
-});
+// We use this component to simulate the whole flow of
+// changing a value and formatting on blur.
+class TestComponent extends React.Component {
+  static displayName = 'TestComponent';
+  static propTypes = {
+    id: PropTypes.string,
+    value: PropTypes.objectOf(PropTypes.string).isRequired,
+    onChange: PropTypes.func,
+    selectedLanguage: PropTypes.string.isRequired,
+  };
+  static defaultProps = {
+    id: 'some-id',
+    name: 'some-name',
+    value: {
+      en: 'Oh no, oh no, oh no. no',
+      fr: "Calisse d'ostie de tabarnak",
+    },
+    selectedLanguage: 'en',
+  };
 
-describe('rendering', () => {
-  let wrapper;
-  let props;
-  describe('with minimal props', () => {
-    beforeEach(() => {
-      props = createTestProps();
-      wrapper = shallow(<LocalizedTextInput {...props} />);
-    });
-    it('should match snapshot', () => {
-      expect(wrapper).toMatchSnapshot();
+  state = {
+    value: this.props.value || {},
+    selectedLanguage: this.props.selectedLanguage || '',
+  };
+
+  handleChange = event => {
+    event.persist();
+    this.setState({
+      value: {
+        ...this.state.value,
+        [event.target.language]: event.target.value,
+      },
     });
+  };
+
+  render() {
+    return (
+      <LocalizedTextInput
+        {...this.props}
+        onChange={this.props.onChange || this.handleChange}
+        value={this.state.value}
+        selectedLanguage={this.state.selectedLanguage}
+      />
+    );
+  }
+}
+
+const renderLocalizedTextInput = (props, options) => ({
+  ...render(<TestComponent {...props} />, options),
+});
```

Comment:
Nitpick: No need to create the object for now. We can do

```
const renderLocalizedTextInput = (props, options) =>
  render(<TestComponent {...props} />, options)
```

We might actually not use this function at all and just always do `render(<TestComponent {...props} />, options)` as this case here is pretty standard and boilerplate-free. It's up to you :)